crates/dtrace_probe/src/main.rs

```diff
@@ -0,0 +1,52 @@
+use composer::api::{Client, Event};
+use dtrace::{DTrace, ProgramStatus};
+use eyre::Result;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "DTrace Probe", about = "Options for DTrace probe")]
+struct Opt {
+    #[structopt(short, long)]
+    server_address: Option<String>,
+
+    #[structopt(short, long)]
+    process_id: u32,
+}
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+
+    let opt = Opt::from_args();
+
+    let client = match opt.server_address {
+        Some(address) => Client::new(address)?,
+        None => Client::try_default()?,
+    };
+
+    let mut dtrace = DTrace::new()?;
+
+    let pid = opt.process_id;
+    let options = [("bufsize", "1k")];
+    dtrace.execute_program(&format!("syscall:::entry /pid == {pid}/ {{}}"), &options)?;
+
+    let mut probes = Vec::default();
```

Comment:
Aha! So I was testing this probe locally and it just kept crashing my pc. I then tried to change `wait_and_consume` to return a `Vec` instead of appending to one (just as in the linked suggestion) and that fixed it. .)

I'm not sure *why* though. ./

Reply:
Interesting... dtrace also crashed my pc several times. It's very non-deterministic and I haven't figured out the causes yet. I will continue looking.